File: app/src/main/java/com/dewonderstruck/apps/ashx0/db/PSAppInfoDao.java
package com.dewonderstruck.apps.ashx0.db;

import com.dewonderstruck.apps.ashx0.viewobject.PSAppInfo;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface PSAppInfoDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(PSAppInfo PSAppInfo);

    @Query("DELETE FROM PSAppInfo")
    void deleteAll();
}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/db/PSAppInfoDao.kt
package com.dewonderstruck.apps.ashx0.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.dewonderstruck.apps.ashx0.viewobject.PSAppInfo

@Dao
interface PSAppInfoDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(PSAppInfo: PSAppInfo?)

    @Query("DELETE FROM PSAppInfo")
    fun deleteAll()
}